### DISCUSSION TITLE
airism camera f value

### DISCUSSION THREAD

**Discussion by Flautdn** [microsoft/AirSim#4212] [General]
How to know the f value for cameras put on airsim vehicle or environment

**Comment by Flautdn** [comment]
Here f refers the focal length which will give me image co-ordinates to real co-ordinates values.

Cam you explain how to get the focal length value for the camera.

**Comment by zimmy87** [comment]
I'm not entirely familiar with f values/f-stop/focal length, but for the default camera placed on vehicles, there is no depth of field effect applied (the default Unreal Camera Actor essentially has an infinite depth of field: everything is rendered in focus), so the I don't think there is a valid f-stop or focal length (perhaps it's 0 or infinity? I'm not sure how the math would work for an infinite depth of field). If you look in the Blueprint implementation of the PIPCamera class (our custom child of the CameraActor class), the Aperture/F-stop property is disabled:

![image](https://user-images.githubusercontent.com/5205889/146156994-dd55d913-503a-4bb6-a6fe-15688b083c6e.png)

To obtain a valid f-stop or focal length, I believe you would either need to enable the [depth of field post processing effect](https://docs.unrealengine.com/4.26/en-US/RenderingAndGraphics/PostProcessEffects/DepthOfField/) or convert the existing PIPCamera class to be child of the [Cinematic Camera Actor](https://docs.unrealengine.com/4.27/en-US/RenderingAndGraphics/PostProcessEffects/DepthOfField/CinematicDOFMethods/). There is a PR for converting the PIPCamera [here](https://github.com/microsoft/AirSim/pull/3949); I've tested this PR and it does work, but there is some outstanding feedback we're hoping to address before merging it into master.

**Comment by Flautdn** [comment]
Hi

   Thanks
I want to know the below values for the center camera on drone


[image: Screenshot from 2021-12-15 15-11-53.png]

**Comment by zimmy87** [comment]
The center camera still uses the default camera settings, so it will have all the same values as a regular PIPCamera.

**Comment by Flautdn** [comment]
I require an obstacle map from the unreal environment which describes the unreal map grid with free and obstacles.

What are the available methods I can get such obstacle data from the umap.


Thanks

**Comment by zimmy87** [comment]
I'm not very familiar with getting obstacle data from a umap. It sounds like RTABMAP may be a potential solution for your needs. There's been some work done using RTABMAP with AirSim (see https://www.youtube.com/watch?v=60kSTv1Nndk), but I don't think there's documentation or a tutorial for doing this yet, so if you want to go this route, it will probably require some research and experimentation on your end.

**Comment by Flautdn** [comment]
how to set cameras on drone; orientation values in python apis. 
I am in airsim 1.3.1 when I set vehicle camera settings in settings.json It crashes.


Thank you.

**Comment by zimmy87** [comment]
There is a `simSetCameraPose` API you can call from python to change a camera's orientation.

**Reply by sandilyasg** [reply]
@zimmy87 does this work with Unity? I am trying to do this but it crashes
